src/http/server.cr

```diff
@@ -217,8 +243,50 @@ class HTTP::Server
     def bind_ssl(host : String, context : OpenSSL::SSL::Context::Server) : Socket::IPAddress
       bind_ssl(host, 0, context)
     end
+
+    # Creates an `OpenSSL::SSL::Server` and adds it as a socket.
+    #
+    # The SSL server wraps a `TCPServer` listenting on an unused port on *host*.
+    #
+    # ```
+    # server = HTTP::Server.new { }
+    # context = OpenSSL::SSL::Context::Server.new
+    # context.certificate_chain = "openssl.crt"
+    # context.private_key = "openssl.key"
+    # address = server.bind_ssl Socket::IPAddress.new("127.0.0.1", 8000), context
+    # ```
+    def bind_ssl(address : Socket::IPAddress, context : OpenSSL::SSL::Context::Server) : Socket::IPAddress
+      bind_ssl(address.address, address.port, context)
+    end
   {% end %}
 
+  # Parses a socket configuration from *uri* and adds it to this server.
+  # Returns the effective address it is bound to.
+  #
+  # ```
+  # server = HTTP::Server.new { }
+  # server.bind("tcp://localhost:8080") # => Socket::IPAddress.new("localhost, 8080")
+  def bind(uri : String) : Socket::Address
+    bind(URI.parse(uri))
+  end
+
+  # ditto
+  def bind(uri : URI) : Socket::Address
+    case uri.scheme
+    when "tcp"
+      bind_tcp(Socket::IPAddress.parse(uri))
+    when "unix"
+      bind_unix(Socket::UNIXAddress.parse(uri))
+    when "ssl"
+      address = Socket::IPAddress.parse(uri)
+      context = OpenSSL::SSL::Context::Server.parse(HTTP::Params.parse(uri.query || ""))
+
+      bind_ssl(address, context)
```

Comment:
Checking for `flag?(:without_openssl)` is missing here. At least with a duplication of the Unsupported socket type message or something more specific.